messaging/src/main/java/org/axonframework/eventhandling/async/MetaDataSequencingPolicy.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2010-2021. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.async;
+
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.eventhandling.EventMessage;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * A {@link SequencingPolicy} implementation that extracts the sequence identifier from the event message metaData based
+ * on a given metaData key.
```

Comment:
Massive nit, but here goes:
```suggestion
 * A {@link SequencingPolicy} implementation that extracts the sequence identifier from the {@link EventMessage}'s {@link org.axonframework.messaging.MetaData}, based
 * on a given {@code metaDataKey}.
```

Reply:
I'd also describe what the default sequence identifier is in the absence of the metadata key entry, by the way.